src/modules/rpcModule.ts

```diff
@@ -212,31 +209,24 @@ export class RpcModule extends SuiClient {
         return simulateRes
       }
 
-      // If useDevInspect is false, manually construct the transaction for simulation.
-      const inputs = tx.blockData.inputs.map((input) => {
-        const { type, value } = input
-        if (type === 'object') {
-          return Inputs.SharedObjectRef({
-            objectId: value,
-            initialSharedVersion: 0,
-            mutable: true,
-          })
-        }
-        return value
+      const copy = Transaction.from(tx)
+
+      Transaction.from(tx).addSerializationPlugin((data, options, next) => {
```

Comment:
this should be tested to make sure it works the same way as the previous implementation